gateway/src/main/java/org/georchestra/gateway/filter/headers/providers/GeorchestraUserHeadersContributor.java

```diff
@@ -43,26 +45,28 @@ public class GeorchestraUserHeadersContributor extends HeaderContributor {
-                        add(headers, "sec-email", mappings.getEmail(), user.map(GeorchestraUser::getEmail));
-                        add(headers, "sec-firstname", mappings.getFirstname(), user.map(GeorchestraUser::getFirstName));
-                        add(headers, "sec-lastname", mappings.getLastname(), user.map(GeorchestraUser::getLastName));
-                        add(headers, "sec-tel", mappings.getTel(), user.map(GeorchestraUser::getTelephoneNumber));
+                        add(headers, SEC_USERID, mappings.getUserid(), user.map(GeorchestraUser::getId));
+                        add(headers, SEC_USERNAME, mappings.getUsername(), user.map(GeorchestraUser::getUsername));
+                        add(headers, SEC_ORG, mappings.getOrg(), user.map(GeorchestraUser::getOrganization));
+                        add(headers, SEC_EMAIL, mappings.getEmail(), user.map(GeorchestraUser::getEmail));
+                        add(headers, SEC_FIRSTNAME, mappings.getFirstname(), user.map(GeorchestraUser::getFirstName));
+                        add(headers, SEC_LASTNAME, mappings.getLastname(), user.map(GeorchestraUser::getLastName));
+                        add(headers, SEC_TEL, mappings.getTel(), user.map(GeorchestraUser::getTelephoneNumber));
 
                         List<String> roles = user.map(GeorchestraUser::getRoles).orElse(List.of());
 
-                        add(headers, "sec-roles", mappings.getRoles(), roles);
+                        add(headers, SEC_ROLES, mappings.getRoles(), roles);
 
-                        add(headers, "sec-lastupdated", mappings.getLastUpdated(),
+                        add(headers, SEC_LASTUPDATED, mappings.getLastUpdated(),
                                 user.map(GeorchestraUser::getLastUpdated));
-                        add(headers, "sec-address", mappings.getAddress(), user.map(GeorchestraUser::getPostalAddress));
-                        add(headers, "sec-title", mappings.getTitle(), user.map(GeorchestraUser::getTitle));
-                        add(headers, "sec-notes", mappings.getNotes(), user.map(GeorchestraUser::getNotes));
-                        add(headers, "sec-ldap-remaining-days", Optional
+                        add(headers, SEC_ADDRESS, mappings.getAddress(), user.map(GeorchestraUser::getPostalAddress));
+                        add(headers, SEC_TITLE, mappings.getTitle(), user.map(GeorchestraUser::getTitle));
+                        add(headers, SEC_NOTES, mappings.getNotes(), user.map(GeorchestraUser::getNotes));
+                        add(headers, SEC_LDAP_REMAINING_DAYS, Optional
                                 .of(user.isPresent() && user.get().getLdapWarn() != null && user.get().getLdapWarn()),
                                 user.map(GeorchestraUser::getLdapRemainingDays));
+                        add(headers, SEC_EXTERNAL_AUTHENTICATION,
+                                Optional.of(user.isPresent() && user.get().getIsExternalAuth()), Optional.of("true"));
```

Comment:
Is the last argument supposed to be the default value ? Won't it make sense to have it `optional.of("false")` by default ?

Reply:
'Optional.of(user.isPresent() && user.get().getIsExternalAuth())'
suppose that there is a user at first then it will read default value for getIsExternalAuth() (supposed false).
But we need to take into consideration the case when getIsExternalAuth() return true.

Optional.of("true") is like saying always send this attribute in header whenever the case.

Thanks

Reply:
ok, my misunderstanding was on the nature of the parameter: I thought it was meant to provide a default value, it is just to indicate whether the header is enabled or not.

Makes more sense to me then.